### ISSUE TITLE
Integrate UMU-Launcher for a Native-Like Windows Gaming Experience

### ISSUE THREAD

**Issue by RaulKong898** [bottlesdevs/Bottles#3537]
### Tell us the problem or your need

Currently, running Windows games on Linux with Bottles, while functional, can sometimes lead to compatibility issues and performance limitations due to the inherent challenges of emulation.  This can result in a less than ideal gaming experience for users.

### Describe the solution you'd like

Integrate UMU-Launcher into Bottles. UMU-Launcher leverages Steam Runtime and Proton, offering a containerized environment optimized for running Windows games. This integration would provide:

Enhanced Compatibility: By utilizing Proton, compatibility with a wider range of Windows games would be significantly improved.
Increased Performance: The optimized runtime environment provided by UMU-Launcher would lead to smoother and faster gaming performance, minimizing performance overhead typically associated with emulation.
Unified Experience: Users would enjoy a seamless experience running games from various platforms (Steam, Epic Games Store, GOG, etc.) within Bottles, simplifying game management.
This would essentially allow Bottles to run Windows games with a level of compatibility and performance approaching that of native Linux games.

### Other solutions?

While other solutions exist for improving Windows game compatibility on Linux, such as using different Wine versions or tweaking configurations, they often require manual intervention and technical expertise. Integrating UMU-Launcher offers a more streamlined and user-friendly solution, automating many of the complexities involved in optimizing game performance.

### Additional context and references

UMU-Launcher GitHub repository: <https://github.com/Open-Wine-Components/umu-launcher>
UMU-Launcher documentation: [https://github.com/Open-Wine-Components/umu-launcher/blob/main/README.md](https://www.google.com/url?sa=E&source=gmail&q=https://github.com/Open-Wine-Components/umu-launcher/blob/main/README.md)

This feature would greatly enhance the Bottles gaming experience, attracting more users to the platform and solidifying its position as a leading solution for Windows gaming on Linux.

Submitted by: Raul Popescu (aka Kong)

**Comment by Arcitec** [comment]
The creator of Bottles is one of the people who have write access to the UMU organization so I am 100% sure the plan is to add support for it later. I hope it's soon because it's *the* best way to run games. And Wine-GE has been abandoned for 9 months since Proton-GE with UMU is the new solution.

https://github.com/Open-Wine-Components

https://github.com/GloriousEggroll/wine-ge-custom/releases/tag/GE-Proton8-26

So right now we don't have a good runner for Bottles:

- Outdated Proton, doesn't work for many of my games.
- New Proton, but with the bad way of running it which has some stability issues.

UMU is desperately needed. :)

**Comment by SeongGino** [comment]
Honestly the most important part for me is having the fshack *and* the integrated gstreamer and video fixes integrated--two features that no other available runner seems to have access too.

Which is unfortunate! Because Bottles has become my go-to, but the current implementation of Proton support leaves *much* to be desired, and I've been having to use Lutris in the meanwhile just to keep up-to-date with this forced change in GE updates... which while functional, is nowhere near as polished or friendly or functional as the Bottles experience.

I really, *really* hope that better Proton support is coming soon. Fingers crossed?

**Comment by SeongGino** [comment]
@Arcitec I appreciate the thought, but I simply *don't like* Heroic's design and it's simply not the same workflow at all.

**Comment by Arcitec** [comment]
@SeongGino I agree that Bottles has a better interface. And I used to dislike Heroic's UI design in the past.

But after getting used to it, it's totally fine, almost as good as Bottles. Even better in some ways. Bottles really wins in three areas: Easier installing dependencies (winetricks is ugly), some settings are easier to toggle via Bottles settings, and Bottles makes it possible to create multiple shared environments (Heroic only supports one shared environment for the games that don't use per-game prefixes).

Of course, I really look forward to the Electron-based Bottles Next:

https://usebottles.com/posts/2023-10-05-bottles-next-a-new-chapter/

That GUI is truly far better than anything from Bottles, Lutris or Heroic right now.

Anyway, for Heroic, you really should explore its settings, if you want to use **the only UMU-Based launcher** right now:

- It has lots of styles/themes you can switch between.
- Enable the experimental toggles to enable the new design.
- Learn a bit and get used to its configuration UI.

![image](https://github.com/user-attachments/assets/c48ea3d7-523e-45d2-8e70-5d5f852d952c)

![image](https://github.com/user-attachments/assets/775f8534-ad6d-4772-827d-9681e86381b4)

![image](https://github.com/user-attachments/assets/5ba02c9d-4320-410b-89b4-2e062eaba2b3)

**Comment by Arcitec** [comment]
@mirkobrombin Regarding Bottles, even though you're clearly busy with Bottles Next, I wonder if it would be possible to change this into a dropdown instead?

## Steam Runtime

- Disabled
- Steam Runtime
- UMU Launcher

And if UMU is picked, Bottles downloads and auto-updates it internally.

From what I understand, UMU Launcher is pretty much a drop-in replacement for Steam Runtime.

Would this be feasible?

![image](https://github.com/user-attachments/assets/3728ddc0-45b6-42c2-8976-324a6a3499ed)

**Comment by SeongGino** [comment]
@Arcitec Good for you I guess? My original point still stands though, as is my lack of desire to use an alternative that doesn't have the same workflow (as well as a boatload of other problems when *not* being used as an EGS alternative).

**Comment by Arcitec** [comment]
UMU support in Bottles would be the optimal solution of course.

**Comment by Qronikarz** [comment]
The main developer of Bottles is already a part of the UMU and OpenWineComponents. I am sure it is already in works, but it probably won't be soon (maybe even only when Bottles Next drops)
Also, there is a blog post on Bottles blog addressing this issue: https://usebottles.com/posts/2024-09-24-umu-next/

> I wonder if it would be possible to change this into a dropdown instead?

Quote from Bottles blog:
> Implementing UMU requires more complex design work, as UMU needs to create and manage these bottles (wineprefix), but the architectural challenges don’t stop there. UMU is tightly connected to the Wine components that we define as “legacy” in Bottles—tools that aren’t implemented within Bottles’ architecture. This includes things like DLL overrides, drive management, Windows version control, virtual desktops, fullscreen capture, mouse warp, DPI settings, rendering modes, process management, or even dependency management itself, which is handled by Winetricks in UMU and other platforms. In Bottles, we have a dedicated infrastructure and a custom syntax to simplify and streamline the creation of new dependencies and installers.

Of course, that depends how you interpret it, but to me it seems like it can't be turned into a simple dropdown switch.

**Comment by Amirmohammadomidi** [comment]
Would love to know the state of bottles next. Integrating UMU would be highly beneficial.

**Comment by TheEpicNoobZilla** [comment]
Yeah UMU-Launcher would be probably best solution since wine-ge is no more, because kron4ek is not ideal solution (although it works with 99% programs i used under bottles). Biggest issue currently with lack of wine-ge are GOG Galaxy games, since default wine from flatpak (wine 10 at time of writing) works with launcher, but not with Witcher 3, while Cyberpunk 2077 suffers from constant audio crackling, using kron4ek's build of staging or tkg causes the launcher to crash, but fixes games mentioned above, dropping to wine-ge 8.26 fixes Witcher 3, but breaks Cyberpunk even more since now it also adds constant stuttering, there are also games like Witcher 2 which without staging, tkg or ge it locks itself to 24-30 FPS. I know with GOG Galaxy i should probably use HGL, but i prefer to have one wine prefix manager and i don't like lack of prefix isolation in heroic or lutris (ie the programs can and will write to my ~/ folders like Borderlands 2 writing to my ~/Documents instead of keeping it inside prefix)

**Comment by jarrard** [comment]
Pretty much ANY game that uses G-Streamer to playback video content doesn't work in ANY of my bottles and I've tried many proton/wine builds. 

Meanwhile exact same builds work under Lutris and even Heroic.  It's clear bottles has major issue linking to system video playback resources, and I'm using NATIVE build not flatpak!   

Basically have had to stop using Bottles now, it just too far behind, it doesn't even allow CUSTOM options for its gamescope settings which is VERY limited.

**Comment by mirkobrombin** [comment]
> Pretty much ANY game that uses G-Streamer to playback video content doesn't work in ANY of my bottles and I've tried many proton/wine builds.
> 
> Meanwhile exact same builds work under Lutris and even Heroic. It's clear bottles has major issue linking to system video playback resources, and I'm using NATIVE build not flatpak!
> 
> Basically have had to stop using Bottles now, it just too far behind, it doesn't even allow CUSTOM options for its gamescope settings which is VERY limited.

All those platforms doesn't just run wine + your app, there are so many differences between them. Just telling "it does not work here but work there" does not help debugging things. There are examples of apps not working on any of them but working on Bottles, so.. do we have to stop using those platforms because of this?

If you actually want this fixed, you need to provide actionable data rather than complaints. We need to know if you have opened a dedicated issue for your specific GStreamer problem with the necessary verbose logs attached, rather than just commenting here. 

Debugging Wine is complex and requires knowing your exact environment: which specific Linux distribution are you running? What GPU are you using, and are you on Nvidia proprietary drivers? These factors critically change how video backends and Gamescope behave.
You explicitly mentioned using the native build as if that should guarantee stability, but it is often the opposite. 

While for sure Flatpak provides a standardized runtime environment, using the native package means Bottles is entirely at the mercy of your specific system libraries and configuration. It is not magic; if your distribution's GStreamer plugins or 32-bit dependencies differ even slightly from what the runner expects, things will break. Regarding Gamescope, if a specific argument you need is missing, that is a standard feature request you should submit. Claiming the software is "too far behind" without providing logs, hardware specs, or constructive feedback does not help anyone fix the issue, it just increases the already extensive toxicity we have in the community. 

PS. For gamescope.. is there an issue already? Will you open it or ranting on a random issue is your way of helping?

**Comment by CaptainCoward** [comment]
@mirkobrombin i don't know if i can be actually helpful here but i also ran into Gstreamer issues lately. I tried pretty much all runners but none seems to work. 
The example here would be Resident Evil (HD).  The game would start but then end up having a blackscreen. When i start it from terminal i get lots of messages that gstreamer is missing a plugin and can't load certain files/plugins.
And example would be

`(wine:1138): GStreamer-WARNING **: 11:03:38.638: Failed to load plugin '/usr/lib/x86_64-linux-gnu/gstreamer-1.0/libgstplayback.so': /usr/lib/x86_64-linux-gnu/gstreamer-1.0/libgstplayback.so: wrong ELF-Class: ELFCLASS64`

 From what i read in the internet the issue seems to be related to wmv file videos.  The terminal also mentions that like this
`
Missing decoder: Windows Media Video 9 (video/x-wmv, wmvversion=(int)3, format=(string)WVC1, width=(int)1920, height=(int)1080, codec_data=(buffer)270000010fe3fe3bf21b8a3bf886f1c0efd09bcc4adf5dc00000010e5a67f840, framerate=(fraction)2500000/83333)`


I tried just installing Windows Media player into the wineprefix but it's not pssoible since the current version is "newer".

**Comment by mirkobrombin** [comment]
UMU integration is implemented in Bottles Next and is not planned for the legacy Bottles codebase. Proton operations are routed through `umu-run` in next-core:

https://github.com/bottlesdevs/next-core/blob/main/src/runner/proton.rs
